File: library/src/main/ts/libraryBrs/src/app/core/services/rest/libraries.service.ts
import { Injectable } from '@angular/core';
import { BehaviorSubject, Observable } from 'rxjs';
import { Response } from './api/api';
import { RestService } from './rest.service';
const URL = "api/libraries";

@Injectable()
export class LibrariesService {
  public libraries: BehaviorSubject<[]> = new BehaviorSubject([]);

  constructor(private restService: RestService) { }

  getContact(): Observable<Response<any>> {
    return this.restService.get(`${URL}/contact`);
  }

  getTerms(): Observable<Response<any>> {
    return this.restService.get(`${URL}/terms`);
  }
}

File: library/src/main/dart/mobile/reader/lib/services/librariesService.dart
import 'dart:collection';

import 'package:flutter/cupertino.dart';
import 'package:reader/services/api/api.dart';
import 'package:reader/services/restService.dart';

class LibrariesService extends ChangeNotifier {
  final String _url = "api/libraries";
  final List<Library> _libraries = [];

  UnmodifiableListView<Library> get items => UnmodifiableListView(_libraries);

  void getContact() {
    RestService rest = RestService();
    rest.get<dynamic>(
        path: "${_url}/contact", onSuccess: onSuccessGetContact);
  }

  void onSuccessGetContact(dynamic object) {
    print(object);
    Response response = Response.fromJson(object);
    notifyListeners();
  }

  void getTerms() {
    RestService rest = RestService();
    rest.get<dynamic>(
        path: "${_url}/terms", onSuccess: onSuccessGetTerms);
  }

  void onSuccessGetTerms(dynamic object) {
    print(object);
    Response response = Response.fromJson(object);
    notifyListeners();
  }
}
